plugins/inputs/webhooks/webhooks.go

```diff
@@ -70,16 +73,11 @@ func (wb *Webhooks) Gather(_ telegraf.Accumulator) error {
 	return nil
 }
 
-func (wb *Webhooks) Listen(acc telegraf.Accumulator) {
-	r := mux.NewRouter()
-
-	for _, webhook := range wb.AvailableWebhooks() {
-		webhook.Register(r, acc)
-	}
-
-	err := http.ListenAndServe(fmt.Sprintf("%s", wb.ServiceAddress), r)
-	if err != nil {
-		acc.AddError(fmt.Errorf("E! Error starting server: %v", err))
+func (wb *Webhooks) listen(ln net.Listener, acc telegraf.Accumulator) {
```

Comment:
Lets call this `serve` since we are already listening.

Reply:
Or just inline it in the `Start` func.